_posts/2021-12-21-gs_00005_environment_variables.md

```diff
@@ -0,0 +1,223 @@
+            -new-tab -url https://github.com/bitsandvolts/getting_started &
+
+Add the correct privilege (this will be explained in the next post!):
+
+    $ chmod +x open_firefox.sh
+
+In our last post, we have already seen that it cannot be run the following:
+
+    $ open_firefox.sh
+
+The reason is that the script is not stored at a location that is part of the
+'PATH' variable, so it cannot be found. Solution number one was to add the
+script's path, which is your home directory, to the 'PATH' variable. This can be
+done by using the 'export' command:
+
+    $ export PATH=$HOME
+
+Here, you set the value of 'PATH' to the content of the 'HOME' variable. If you
+display the content,
+
+    $ echo $PATH
+
+you should see that it changed.
+
+<figure class="img-center" style="width:90%">
+  <img src="/assets/images/gs_00005_export.jpg" alt="">
+</figure>
+
+Now, try to run the script again as before:
+
+    $ open_firefox.sh
+
+<figure class="img-center" style="width:90%">
+  <img src="/assets/images/gs_00005_firefox.jpg" alt="">
+</figure>
+
+The script can be found, but now we get the error that firefox cannot be found.
+Hmm ... It is the same script as we used in the last post and there it worked.
+What can be wrong? Can you guess? We have overwritten the 'PATH' variable so
+that it solely consists of your home directory. Since firefox is not located
+there, it cannot be found. By the way, none of the Linux commands we've learned
+works. Have a try:
+
+    $ ls        # does not work
+    $ /bin/ls   # call by absolute path works
+
+Close and reopen the terminal. Again, print the value of the 'PATH' variable:
+
+    $ echo $PATH
+
+You should note that it has its original value. So, using the export command as
+we did before changes the value only for the current terminal session. A new
+session will reinitialize all values. Now, we use the 'export' command slightly
+different:
+
+    $ export PATH=$HOME:$PATH
+
+What does that do? Again we set the value of the 'PATH' variable. We've already
+mentioned that the directories in the 'PATH' variable are separated by a colon.
+So, we set the value to the home folder separated by a colon to the previous
+value of the 'PATH' variable. We simply _add_ another directory. This time, the
+script will work:
+
+    $ open_firefox.sh
+
+# Permanent change
+
+There is one issue left. The setting stays only for the current session. What if
+you want to have it changed permanently? We do not want to dive too deep into
+this for now, but we should mention that, when you open a terminal, several
+scripts are executed automatically. For example, ~/.bashrc (reminder: ~ stands
+for your home directory) is one of them.
+
+    $ nano ~/.bashrc
+
+and add the 'export' command to the end of the file.
+
+    $ export PATH=$HOME:$PATH
+
+Save and close the file. Then, reopen your terminal and display the 'PATH'
+variable.
+
+    $ echo $PATH
+
+Your home directory should be the first one.
```

Comment:
Maybe some kind of "outro" like this?
```suggestion
Your home directory should be the first one. Now, you can instantly run the `open_firefox.sh` script from every terminal you open, and from any location. Keep the PATH variable in mind the next time you write a script, it can save you a lot of time!
```

Reply:
Just a note for me: Outro is good. Just would add the hint that its not a good solution to always add the path to the PATH variable

Reply:
better?